### ISSUE TITLE
Create Linux binary

### ISSUE THREAD

**Issue by petebankhead** [qupath/qupath#2]
QuPath should (i.e. has in the past) compile fine under Linux, with the required native libraries.

However, no distribution for Linux is available under _Releases_ yet.

This should happen, but requires gaining access to a suitable Linux machine soon to arrange and test it...

**Comment by petebankhead** [comment]
Linux binaries are only available for the very first release (v0.0.1).

There hasn't yet been any audible demand for updated binaries... but they should at least be provided for v0.1.

Ideally these would use OpenJDK and OpenJFX... although so far it has proven easier to create them using Oracle's JDK.

Any Linux users who would like to see Linux binaries are invited to comment here...

**Comment by petebankhead** [comment]
Eventually I managed to create a new Linux download for [v0.0.6](https://github.com/qupath/qupath/releases/tag/v0.0.6) using OpenJDK and OpenJFX.

Still, two troubles remain: one minor, one major.

The minor trouble is that (at least on Ubuntu 16.04) the build process is not entirely seamless, and ```jfxrt.jar``` needs to be manually copied into the distribution at the end (perhaps because it's not included in OpenJDK, but rather only a symbolic link is made with installing ```openjfx``` with Synaptic?).

The major trouble is that OpenSlide continues not to be distributed in a ready-to-run fashion (as mentioned [here](https://github.com/qupath/qupath/wiki/Installing-QuPath#linux)).  Indeed, I'm not sure if it will work on anyone's machine other than the one I was using.

It is likely clear that my Linux use and experience is limited.  The suggestions of someone who is more familiar with this would be very welcome.  I think one of the following is required:

* Inclusion of OpenSlide and all its dependencies, precompiled (in the same way as QuPath provides for Windows and macOS), or
* A method of ensuring the Java library path in the packaged version of QuPath includes all the places that would be required to find a local installation of OpenSlide

In the second case, the user would still be asked to download and install it... but if this step goes smoothly, it can be expected to work.

I plan to investigate this more in the future, but it may take some time...

**Comment by petebankhead** [comment]
This situation is hopefully much improved in v0.0.7.

OpenSlide and OpenCV were both recompiled, and I attempted to ensure all non-standard, required dependencies are now included - with an appropriate ```rpath``` setting so that they may find one another.

Most of the steps taken are documented under the ['Maven' directory](https://github.com/qupath/qupath/tree/master/maven).

The Linux version 0.0.7 was created using Ubuntu 16.04 running through VirtualBox on an iMac.  I also tested the compiled QuPath on a clean installation of Fedora 25, also through VirtualBox.  Both were able to run QuPath, including commands depending on OpenSlide and OpenCV - although things did run a bit more smoothly on Ubuntu (e.g. the *Brush* tool misbehaved on Fedora).

Additionally, the Linux version is created using [OpenJDK](http://openjdk.java.net) and [OpenJFX](http://openjdk.java.net/projects/openjfx/) - as opposed to the [Oracle JDK](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) used for Windows and Mac.
